doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,574 @@
+# Feature Name
+RDMA (Remote Direct Memory Access) over Converged Ethernet (ROCEv2) support
+
+# High Level Design Document
+
+#### Rev 0.2
+# Table of Contents
+
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 06/17/2021 |   Venkatesan Mahalingam         | Initial version                   |
+| 0.2 | 07/16/2021 |   Venkatesan Mahalingam         | Addressed review comments                   |
+
+
+# About this Manual
+
+This document introduces the support of ROCEv2 in mgmt-framework,
+this provides the North Bound Interface (NBI) interfaces i.e REST/gNMI/KLISH for the ROCEv2 configurations.
+
+# Scope
+
+This document covers the following,
+
+1) NBI interface to initialize QoS buffers for PFC \
+2) NBI interface to create/delete buffer pool \
+3) NBI interface to create/delete buffer profile \
+4) NBI interface to map/unmap priority group to buffer profile \
+5) NBI interface to set/unset PORT table for default buffer profile creation
+   based on cable length and speed
+6) Unit Testcases
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| RDMA                     | Remote Direct Memory Access         |
+|          ROCEv2            |  RDMA over Converged Ethernet version 2           |
+|          PFC            |  Priority Flow Control|
+
+
+# 1 Feature Overview
+
+ROCE provides the ability to transport Infiniband (IB) protocol traffic using RDMA over lossless converged Ethernet
+across storage or compute nodes.
+ROCEv2 stands for RDMA over Converged Ethernet version 2, this is an extension of ROCE protocol a.k.a routable ROCE,
+This overcomes the limitation of ROCEv1 bounded to a single broadcast domain (VLAN),
+ROCEv2 encapsulates an RDMA transport packet within the Ethernet/IPv4/UDP packet,
+ROVEv2 uses PFC to prevent buffer overflow.
+
+## 1.1 Requirements

```

Comment:
As of now only configuration requirements are mentioned here. Do we have any functional requirements for ROCEv2 feature validation? Request to provide any such details are available. What exactly we are going to validate as part of this feature testing? Please provide.

Reply:
As part of the feature testing, we're simulating traffic congestion using IXIA to test PFC, I think, system test scope is to verify the lossless traffic from server/storage.

Reply:
Do you mean we need to generate L3( IPv4/IPv6 ) traffic using IXIA ports to create congestion and that congestion will be managed by PFC? If Yes, please confirm what is the role of ROCEv2 in that specific case. Thank you.

Reply:
It is actually testing the PFC but the payload could be anything, system test is expected to test the IB traffic from server.

Reply:
What exactly IB traffic means? How do we generate IB traffic? What kind of servers to be used to generate such traffic? Please clarify.